components/camera/extra.go

```diff
@@ -0,0 +1,22 @@
+package camera
+
+import "context"
+
+// Extra is the type of value stored in the Contexts.
+type (
+	Extra map[string]interface{}
+	key   int
+)
+
+var extraKey key
+
+// NewContext returns a new Context that carries value Extra.
+func NewContext(ctx context.Context, e *Extra) context.Context {
+	return context.WithValue(ctx, extraKey, e)
+}
+
+// FromContext returns the Extra value stored in ctx, if any.
+func FromContext(ctx context.Context) (*Extra, bool) {
+	u, ok := ctx.Value(extraKey).(*Extra)
```

Comment:
I tested and this does break filtered cameras, as expected. Instead of setting a [context value](https://github.com/viamrobotics/rdk/blob/bc6ddd1b2e1b93bf1c477df4fa52b29617aff37e/components/camera/collectors.go#L95) with a placeholder `FromDMContextKey`, would we do something like:

```
var extra Extra
extra = make(map[string]interface{})
extra[data.FromDMString] = true
ctx = NewContext(ctx, &extra)
```

and then update the modules/documentation using it (e.g. Eliot's [filtered camera](https://github.com/erh/filtered_camera/blob/344fb3b468e5d67288a530d2a39168cced83232f/cam.go#L201-L204)) to use `FromDMString`?

In order to not break things, could we stage this where this PR handles the one-off case that passes through the FromDMContextKey?

Reply:
Done! This PR shouldn't break anything for data. Lmk if it does. I also added `Deprecated:` where I thought it was appropriate.

Reply:
So, the only tricky thing here is that extra maps are of type `map[string]any` in the SDKs but data uses a custom struct for the key. That means `ctx.Value(data.FromDMContextKey{})` will not work, but you should still be able to get the value of the key with `data.FromDMString` since that's passed through.